src/errors.rs

```diff
@@ -52,6 +52,10 @@ pub enum ApiError {
     #[error("request error: {0}")]
     RequestError(#[source] ReqwestError),
 
+    /// Error when building request URL (via reqwest)
+    #[error("request error: {0}")]
+    RequestParseError(#[from] url::ParseError),
+
```

Comment:
Since we don't parse requests, this would be more precise, right?

```suggestion
    /// Error when building request URL (via reqwest)
    #[error("request URL parse error: {0}")]
    RequestUrlParseError(#[from] url::ParseError),

```